CODEWALK.md

```diff
@@ -362,6 +362,86 @@ Protocols (goroutines) communicate through custom channels.
+
+Sectors and sealed sectors (aka replicas) are just flat files on disk. 
+Sealing a sector is a destructive operation on the sector. 
+The process of sealing yields metadata such as the proof/commitment, which is stored is a separate metadata store, not within the replica file.
+
+We intend the sector builder interface to represent a service, abstracting away both policy (e.g. sector packing, scheduling of PoSt calculation) and implementation details. 
+In the future, we would like to able to interface to it via IPC/RPC as well as FFI. 
+
+The sector builder and proofs code is written in Rust partly to ease use of the [Bellman zk-SNARK library](https://github.com/zkcrypto/bellman). 
+The PoRep and PoSt code is under active development. 
+PoRep is integrated with go-filecoin, while the PoST implementation and integration is still in progress (March 2019).
+
+### Building and distribution.
+The Rust code responsible for sectors and proofs is in the [rust-fil-proofs](https://github.com/filecoin-project/rust-fil-proofs) repo. 
+This repo is included in go-filecoin as a Git submodule. 
+The submodule refers to a specific repository SHA hash. 
+The `install-rust-proofs.sh` script, invoked by the `deps` build step of go-filecoin, builds the Rust proofs code and copies binary assets to locations hardcoded in Go interface code.
+
+As an alternative to compiling Rust code locally, the continuous integration server publishes an archive of precompiled binary objects with every successful build of the `rust-fil-proofs/master` branch.
+These releases are identified by the Git submodule SHA. This archive is pushed to the GitHub releases service. 
+When the `FILECOIN_USE_PRECOMPILED_RUST_PROOFS` environment variable is set, `install-rust-proofs.sh` attempts to fetch these assets from GitHub releases 
+and installs them in the same hardcoded locations required by the Go build.
+
+The build or tarball contains:
+- `libfilecoin_proofs.a` (a static library)
+- `libfilecoin_proofs.h` (corresponding C-header file)
+- `libfilecoin_proofs.pc` (a pkg-config manifest, used to specify linker dependencies)
+- `paramcache` (populates Groth parameter-cache
+- `paramfetch` (fetches Groth parameters from IPFS gateway into cache)
+- `parameters.json` (contains the most recently-published Groth parameters)
+
+### Groth parameters
+The proving algorithms rely on a large binary parameter file known as the Groth parameters. 
+This file is stored in a cache directory, typically `/tmp/filecoin-proof-parameters`. 
+When proofs code changes, the params may need to change.
+
+The `paramcache` program populates the Groth parameter directory by generating the parameters, a slow process (10s of minutes). 
+If the cache directory is already populated, generation is skipped. 
+The `parampublish` program (not part of the precompiled proofs archive) publishes params from the cache directory to a local IPFS node.
+The published CIDs need to be pinned e.g. by a Protocol Labs pinning service. 
```

Comment:
This seems important, but it's meaningless to me. CID's of what? The cache directory? What does it mean to pin a CID?